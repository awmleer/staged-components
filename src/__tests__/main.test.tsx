import {act} from '@testing-library/react'
import {staged} from '..'
import * as React from 'react'
import {useEffect, useState} from 'react'
import * as testing from '@testing-library/react'

export const sleep = (time: number) => new Promise(resolve => setTimeout(resolve, time))

test('provider initialize', async function () {
  const App = staged(() => {
    const [waiting, setWaiting] = useState(true)
    useEffect(() => {
      setTimeout(() => {
        act(() => {
          setWaiting(false)
        })
      }, 300)
    }, [])
    if (waiting) return null
    return () => {
      const [count, setCount] = useState(1)
      return (
        <div>
          <p>{count}</p>
          <button onClick={() => {setCount(count + 1)}}>Change</button>
        </div>
      )
    }
  })
  const renderer = testing.render(
    <App/>
  )
  expect(renderer.asFragment()).toMatchSnapshot()
  await sleep(500)
  testing.fireEvent.click(testing.getByText(renderer.container, 'Change'))
  expect(renderer.asFragment()).toMatchSnapshot()
})
