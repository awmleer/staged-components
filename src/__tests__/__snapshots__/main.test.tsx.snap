// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`provider initialize 1`] = `<DocumentFragment />`;

exports[`provider initialize 2`] = `
<DocumentFragment>
  <div>
    <p>
      2
    </p>
    <button>
      Change
    </button>
  </div>
</DocumentFragment>
`;
